pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -60,6 +64,36 @@ type DefaultPreemption struct {
 	pdbLister policylisters.PodDisruptionBudgetLister
 }
 
+// PreemptionFitError describes a fit error after the pod preemption
+// TODO: Figure out a more decent way to share the type with <core> package. This type is mostly like the `FitError` of <core> package,
+// re-define here simply because we don't want this plugin has a dependency over the <core> package.
+type PreemptionFitError struct {
+	Pod                   *v1.Pod
+	NumAllNodes           int
+	FilteredNodesStatuses framework.NodeToStatusMap
+}
+
+// Error returns detailed information of why the pod failed to fit even after the preemption
+func (f *PreemptionFitError) Error() string {
+	reasons := make(map[string]int)
+	for _, status := range f.FilteredNodesStatuses {
+		for _, reason := range status.Reasons() {
+			reasons[reason]++
+		}
+	}
+
+	sortReasonsHistogram := func() []string {
+		var reasonStrings []string
+		for k, v := range reasons {
+			reasonStrings = append(reasonStrings, fmt.Sprintf("%v %v", v, k))
+		}
+		sort.Strings(reasonStrings)
+		return reasonStrings
+	}
+	reasonMsg := fmt.Sprintf(NoNodeAvailableMsg+": %v.", f.NumAllNodes, strings.Join(sortReasonsHistogram(), ", "))
+	return reasonMsg
+}
+
```

Comment:
lets move FitError to framework/types.go and so we don't need to duplicate this code here.

Reply:
Great suggestion!
Done